### ISSUE TITLE
Tensorrt export fail

### ISSUE THREAD

**Issue by dariogonle** [ultralytics/yolov5#6336]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and found no similar bug report.

### YOLOv5 Component

Export

### Bug

I've used the export.py script to convert yolov5n.pt to tensorrt.
```
python3 export.py --weights yolov5n.pt --include engine
```

But the conversion process fails.

![image](https://user-images.githubusercontent.com/80393891/149969619-cb7632be-b61f-41a0-af50-c51343042589.png)

It says that tensorrt is not installed, but it is:
![image](https://user-images.githubusercontent.com/80393891/149970186-1e4cc298-6609-44c3-810a-911adb891554.png)

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by dariogonle** [comment]
I think the problem is that for using tensorrt it is necessary to force to use the GPU and FP16. Therefore, the correct command should be
```
python export.py --include engine --device 0 --half --weights yolov5n.pt --data data.yaml
```

**Comment by glenn-jocher** [comment]
@dariogonle yes, TRT exports must be run on a GPU backend. Maybe we should have a special assert here for this.

**Comment by glenn-jocher** [comment]
@dariogonle good news 😃! Your original issue may now be fixed ✅ in PR #6340. Based on your feedback this PR adds an informative assert message on attempted TensorRT CPU export:

<img width="1550" alt="Screen Shot 2022-01-18 at 1 49 59 PM" src="https://user-images.githubusercontent.com/26833433/150037540-e876e319-b30b-49f3-8f7e-9372976495e7.png">

To receive this update:
- **[Git](https://github.com/ultralytics/yolov5)** – `git pull` from within your `yolov5/` directory or `git clone https://github.com/ultralytics/yolov5` again
- **[PyTorch Hub](https://pytorch.org/hub/ultralytics_yolov5/)** – Force-reload `model = torch.hub.load('ultralytics/yolov5', 'yolov5s', force_reload=True)`
- **[Notebooks](https://github.com/ultralytics/yolov5/blob/master/tutorial.ipynb)** – View updated notebooks  <a href="https://colab.research.google.com/github/ultralytics/yolov5/blob/master/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov5"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **[Docker](https://hub.docker.com/r/ultralytics/yolov5)** – `sudo docker pull ultralytics/yolov5:latest` to update your image <a href="https://hub.docker.com/r/ultralytics/yolov5"><img src="https://img.shields.io/docker/pulls/ultralytics/yolov5?logo=docker" alt="Docker Pulls"></a>

Thank you for spotting this issue and informing us of the problem. Please let us know if this update resolves the issue for you, and feel free to inform us of any other **issues** you discover or **feature requests** that come to mind. Happy trainings with YOLOv5 🚀!
